plugin/advancedsubscription/src/AdvancedSubscriptionPlugin.php

```diff
@@ -0,0 +1,872 @@
+                }
+            } else {
+                throw new \Exception($this->get_lang('AdvancedSubscriptionNotConnected'));
+            }
+
+            return $isAble;
+        } else {
+            throw new \Exception($this->get_lang('AdvancedSubscriptionIncompleteParams'));
+        }
+
+    }
+
+    /**
+     * Register a user into a queue for a session
+     * @param $userId
+     * @param $sessionId
+     * @return bool|int
+     */
+    public function addToQueue($userId, $sessionId)
+    {
+        $now = api_get_utc_datetime();
+        $pAdvSubQueueTable = Database::get_main_table(TABLE_ADV_SUB_QUEUE);
+        $attributes = array(
+            'session_id' => $sessionId,
+            'user_id' => $userId,
+            'status' => 0,
+            'created_at' => $now,
+            'updated_at' => null,
+        );
+
+        $id = Database::insert($pAdvSubQueueTable, $attributes);
+
+        return $id;
+    }
+
+    /**
+     * Register message with type and status
+     * @param $mailId
+     * @param $userId
+     * @param $sessionId
+     * @return bool|int
+     */
+    public function saveLastMessage($mailId, $userId, $sessionId)
+    {
+        $queueTable = Database::get_main_table(TABLE_ADV_SUB_QUEUE);
+        $attributes = array(
+            'last_message_id' => $mailId,
+        );
+
+        $num = Database::update(
+            $queueTable,
+            $attributes,
+            array('user_id = ? AND session_id = ?' => array($userId, $sessionId))
+        );
+
+        return $num;
+    }
+
+    /**
+     * Check for requirements and register user into queue
+     * @param $userId
+     * @param $sessionId
+     * @param $params
+     * @return bool|string
+     */
+    public function startSubscription($userId, $sessionId, $params)
+    {
+        $result = false;
+        if (!empty($sessionId) && !empty($userId)) {
+            $advSub = self::create();
+            try {
+                if ($advSub->isAbleToRequest($userId, $params)) {
+                    $result = (bool) $advSub->addToQueue($userId, $sessionId);
+                } else {
+                    throw new \Exception($this->get_lang('AdvancedSubscriptionNotMoreAble'));
+                }
+            } catch (Exception $e) {
+                $result = $e->getMessage();
+            }
+        } else {
+            $result = 'Params not found';
+        }
+
+        return $result;
+    }
+
+    /**
+     * Send message for the student subscription approval to a specific session
+     * @param int $studentId
+     * @param int $receiverId
+     * @param string $subject
+     * @param string $content
+     * @param int $sessionId
+     * @param bool $save
+     * @return bool|int
+     */
+    public function sendMailMessage($studentId, $receiverId, $subject, $content, $sessionId, $save = false)
+    {
+        global $_configuration; // @TODO: Add $_configuration['no_reply_user_id'] to configuration file
```

Comment:
There is already a no_reply setting in settings_current. What is it you are looking for?